connaisseur/sigstore_validator.py

```diff
@@ -0,0 +1,78 @@
+import json
+import logging
+import re
+import subprocess  # nosec
+
+from connaisseur.crypto import decode_and_verify_ecdsa_key
+from connaisseur.exceptions import (
+    CosignError,
+    CosignTimeout,
+    NotFoundException,
+    ValidationError,
+)
+
+
+def get_cosign_validated_digests(image: str, pubkey: str):
+    """
+    Gets and processes cosign validation output for a given `image` and `pubkey`
+    and either returns a list of valid digests or raises a suitable exception
+    in case no valid signature is found or cosign fails.
+    """
+    failed, stdout, stderr = invoke_cosign(image, pubkey)
+    logging.info(
+        "COSIGN output for image: %s; RETURNCODE: %s; STDOUT: %s; STDERR: %s",
+        image,
+        failed,
+        stdout,
+        stderr,
+    )
+    digests = []
+    if not failed:
```

Comment:
```suggestion
    if failed == 0:
```

`failed` is `None` if for some reason the process has no return code (not exited yet, but maybe we arrived here by mishandling a timeout?). In this case, `not failed` evaluates to `True`, and we're off on the wrong path. (Explicit is better than implicit!)